quark/subproject.py

```diff
@@ -832,7 +832,9 @@ def _parse_url(self, url):
         self.stamp["sha1"] = fragments.get("sha1")
 
         # Wether to extract the downloaded file (must be either zip, tar.bz2, tar.gz, or tar.xz)
-        self.do_extract = fragments.get("extract", "").lower() == "true"
+        extract = fragments.get("extract", "").lower()
+        self.do_extract = extract in {"true", "zip", "tar"}
```

Comment:
This fails silently (not extracting) for an unsupported format; I'd just make this

```suggestion
        self.do_extract = extract != ""
```

and then handle the various cases down the line, erroring out if there's something we don't understand.